packages/babel-types/scripts/generators/docs.js

```diff
@@ -104,12 +112,170 @@ Object.keys(t.BUILDER_KEYS)
+    const aliases = t.ALIAS_KEYS[nodeType];
+    if (!aliases) continue;
+    for (const alias of aliases) {
+      if (!result.has(alias)) {
+        result.set(alias, []);
+      }
+      const nodeTypes = result.get(alias);
+      nodeTypes.push(nodeType);
+    }
+  }
+  return result;
+}
+const aliasDescriptions = {
+  Binary:
+    "A cover of BinaryExpression and LogicalExpression, which share the same AST shape.",
+  Block: "A cover of BlockStatement and its alike.",
+  BlockParent:
+    "A cover of AST nodes that start an execution context with new [LexicalEnvironment](https://tc39.es/ecma262/#table-additional-state-components-for-ecmascript-code-execution-contexts). In other words, they define the scope of `let` and `const` declarations.",
+  Class:
+    "A cover of ClassExpression and ClassDeclaration, which share the same AST shape.",
+  CompletionStatement:
+    "A statement that indicates the [completion records](https://tc39.es/ecma262/#sec-completion-record-specification-type). In other words, they define the control flow of the program, such as when should a loop break or an action throws critical errors.",
+  Conditional:
+    "A cover of ConditionalExpression and IfStatement, which share the same AST shape.",
+  Declaration:
+    "A cover of any [Declaration](https://tc39.es/ecma262/#prod-Declaration)s.",
+  EnumBody:
+    "A cover of enum bodies defined in [TypeScript](https://www.typescriptlang.org/docs/handbook/enums.html) and Flow.",
+  EnumMember:
+    "A cover of enum membors defined in [TypeScript](https://www.typescriptlang.org/docs/handbook/enums.html) and Flow.",
+  ExportDeclaration:
+    "A cover of any [ExportDeclaration](https://tc39.es/ecma262/#prod-ExportDeclaration)s.",
+  Expression:
+    "A cover of any [Expression](https://tc39.es/ecma262/#sec-ecmascript-language-expressions)s.",
+  ExpressionWrapper:
+    "A wrapper of expression that does not have run semantics.",
+  Flow: "A cover of AST nodes defined for Flow.",
+  FlowBaseAnnotation: "A cover of primary Flow type annotations.",
+  FlowDeclaration: "A cover of Flow declarations.",
+  FlowPredicate: "A cover of Flow predicates.",
+  FlowType: "A cover of Flow type annotations.",
+  For:
+    "A cover of [ForStatement](https://tc39.es/ecma262/#sec-for-statement)s and [ForXStatement](#forxstatement)s.",
+  ForXStatement:
+    "A cover of [ForInStatements and ForOfStatements](https://tc39.es/ecma262/#sec-for-in-and-for-of-statements).",
+  Function:
+    "A cover of functions and [method](#method)s, the must have `body` and `params`. Note: `Function` is different to `FunctionParent`.",
+  FunctionParent:
+    "A cover of AST nodes that start an execution context with new [VariableEnvironment](https://tc39.es/ecma262/#table-additional-state-components-for-ecmascript-code-execution-contexts). In other words, they define the scope of `var` declarations. FunctionParent did not include `Program` since Babel 7.",
```

Comment:
Based on the definition, I think it should also cover `StaticBlock`, which does define a new `var` scope. Note that in Babel 6 `FunctionParent` includes `Program`, but we removed `Program` in https://github.com/babel/babel/pull/5923 because the name `FunctionParent` strongly indicates that it should always be a `Function`, which is not the case for `StaticBlock` and `Program`.

Currently `FunctionParent` and `Function` are equivalent, they include exactly the same AST nodes.

BTW we have several usage of `scope.getFunctionParent() || scope.getProgramParent()`. This depends on the assumption that an AST tree can only have one Program node and all the function node must placed within that program node. This is not true for `module-blocks` proposal.

Reply:
> This is not true for `module-blocks` proposal.

Can you show an example where `scope.getFunctionParent() || scope.getProgramParent()` would break with the `module-blocks` proposal?

Reply:
Ah I missed that module blocks can only be at the top level.

Reply:
No, they can also be nested, but we'll want to consider "the top level" the module block itself.

For example, if we have
```js
let mod = module {
  class A {}
};
```

We want to get this:
```js
let mod = module {
  function _classCallCheck() {}
  let A = function A() {
    _classCallCheck(this);
  };
};
```
and not this:
```js
function _classCallCheck() {}
let mod = module {
  let A = function A() {
    _classCallCheck(this);
  };
};
```

Reply:
Yeah, I just checked module blocks proposal and you are right they can be nested.

> Can you show an example where `scope.getFunctionParent() || scope.getProgramParent()` would break with the module-blocks proposal

```js
function f() {
  (module {
    var x;
  })
}
```

`scope.getFunctionParent() || scope.getProgramParent()` on `var x` will return `function f` instead of the module block.

Reply:
Oh right, thanks